agrona/src/main/java/org/agrona/concurrent/status/AtomicCounter.java

```diff
@@ -372,11 +448,31 @@ public boolean proposeMax(final long proposedValue)
 
     /**
      * Set the value to a new proposedValue if greater than the current value with memory ordering semantics.
+     * <p>
+     * This method is identical to {@link #proposeMaxRelease(long)} and that method should be used instead.
      *
      * @param proposedValue for the new max.
      * @return true if a new max as been set otherwise false.
      */
     public boolean proposeMaxOrdered(final long proposedValue)
+    {
+        return proposeMaxRelease(proposedValue);
+    }
+
+    /**
+     * Set the value to a new proposedValue if greater than the current value.
+     * <p>
+     * This call is not atomic and can suffer from lost updates to race conditions.
+     * <p>
+     * The load has plain memory semantics and the store has release memory semantics.
+     * <p>
+     * The typical use-case is when there is one mutator thread, that calls this method, and one or more reader threads.
+     *
+     * @param proposedValue for the new max.
+     * @return true if a new max as been set otherwise false.
+     * @since 2.1.0
+     */
+    public boolean proposeMaxRelease(final long proposedValue)
```

Comment:
This method needs a CAS loop similar to how `jdk.internal.misc.Unsafe#getAndAddLongRelease` is implemented:
```java
    @ForceInline
    public final long getAndAddLongRelease(Object o, long offset, long delta) {
        long v;
        do {
            v = getLong(o, offset);
        } while (!weakCompareAndSetLongRelease(o, offset, v, v + delta));
        return v;
    }
```

Reply:
Why should the semantics of this method be changed? It wasn't atomic before and now it needs to become atomic; this has performance consequences and isn't needed in case of a single writer multiple reader usage.